leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -178,14 +178,21 @@ protected Object handleInvocation(Object proxy, Method method, Object[] args) th
         try {
             return method.invoke(delegate, args);
         } catch (InvocationTargetException e) {
-            if (e.getCause() instanceof ServiceNotAvailableException
-                    || e.getCause() instanceof NotCurrentLeaderException) {
+            if (e.getTargetException() instanceof ServiceNotAvailableException
+                    || e.getTargetException() instanceof NotCurrentLeaderException) {
                 markAsNotLeading(leadershipToken, e.getCause());
             }
-            throw e.getCause();
+            if (e.getTargetException() instanceof InterruptedException && !(checkIfLeading(leadershipToken))) {
```

Comment:
might be worth a quick comment here explaining that this is for the lock service